pkg/envoy/embedded_envoy.go

```diff
@@ -235,12 +238,14 @@ func startEmbeddedEnvoy(config embeddedEnvoyConfig) (*EmbeddedEnvoy, error) {
 				// Start Envoy again
 				continue
 			case <-envoy.stopCh:
-				log.WithField(logfields.PID, cmd.Process.Pid).Infof("Envoy: Stopping proxy")
+				o.logger.Info("Envoy: Stopping embedded Envoy proxy", logfields.PID, cmd.Process.Pid)
 				if err := envoy.admin.quit(); err != nil {
-					log.WithError(err).WithField(logfields.PID, cmd.Process.Pid).Fatal("Envoy: Envoy admin quit failed, killing process")
```

Comment:
IMO it's ok (even better) to not fatal in these cases. Planned to improve this situation by reporting to a health reporter.